scripts/install_third_party_libs_test.py

```diff
@@ -44,317 +46,370 @@
 
-    def test_tweak_yarn_executable(self):
-        def mock_is_file(unused_filename):
-            return True
+        self.assertTrue(self.check_function_calls['check_call_is_called'])
 
-        def mock_rename(origin_name, new_name):
-            self.assertEqual(origin_name + '.sh', new_name)
-            mock_rename.called = True
-        mock_rename.called = False
-        isfile_swap = self.swap(os.path, 'isfile', mock_is_file)
-        rename_swap = self.swap(os, 'rename', mock_rename)
-        with isfile_swap, rename_swap:
+    def test_tweak_yarn_executable(self) -> None:
+        with self.isfile_true_swap, self.rename_swap:
             install_third_party_libs.tweak_yarn_executable()
-        self.assertTrue(mock_rename.called)
 
-    def test_get_yarn_command_on_windows(self):
-        os_name_swap = self.swap(common, 'OS_NAME', 'Windows')
-        with os_name_swap:
+        self.assertTrue(self.check_function_calls['mock_rename_called'])
+
+    def test_get_yarn_command_on_windows(self) -> None:
+        with self.os_name_windows_swap:
             command = install_third_party_libs.get_yarn_command()
             self.assertEqual(command, 'yarn.cmd')
 
-    def test_get_yarn_command_on_linux(self):
-        os_name_swap = self.swap(common, 'OS_NAME', 'Linux')
-        with os_name_swap:
+    def test_get_yarn_command_on_linux(self) -> None:
+        with self.os_name_linux_swap:
             command = install_third_party_libs.get_yarn_command()
             self.assertEqual(command, 'yarn')
 
-    def test_get_yarn_command_on_mac(self):
-        os_name_swap = self.swap(common, 'OS_NAME', 'Darwin')
-        with os_name_swap:
+    def test_get_yarn_command_on_mac(self) -> None:
+        with self.os_name_mac_swap:
             command = install_third_party_libs.get_yarn_command()
             self.assertEqual(command, 'yarn')
 
-    def test_buf_installation_on_linux(self):
-        check_mock_function_calls = {
-            'url_retrieve_is_called': False,
-            'recursive_chmod_is_called': False,
-            'extractall_is_called': False,
-        }
-
-        class MockZipFile(zipfile.ZipFile):
-            def __init__(self, path, mode): # pylint: disable=unused-argument, super-init-not-called
-                pass
-            def extractall(self, path): # pylint: disable=unused-argument
-                check_mock_function_calls['extractall_is_called'] = True
+    def test_buf_installation_on_linux(self) -> None:
 
-        def mock_url_retrieve(url, filename): # pylint: disable=unused-argument
+        def mock_url_retrieve_linux(url: str, **_kwargs: str) -> None:
             # Check that correct platform is used bufbuild.
             if 'bufbuild' in url:
                 self.assertTrue('Linux-x86_64' in url.split('/')[-1])
-                check_mock_function_calls['url_retrieve_is_called'] = True
+                self.check_function_calls['url_retrieve_is_called'] = True
```

Comment:
I'm not sure this is a good idea. With `check_function_calls` as an instance variable, it'll be easy for us to have a test pass incorrectly because a previous test set `self.check_function_calls['url_retrieve_is_called']` to `True` and the incorrectly-passing test didn't set it to `False`. Why not use `self.swap_with_checks` instead? ditto throughout this file

Since this change would affect a lot of this PR, I didn't review further than this point